pkg/syfs/syfs.go

```diff
@@ -43,20 +44,24 @@ func ConfigDir() string {
 }
 
 func configDir() string {
-	user, err := user.Current()
-	if err != nil {
-		sylog.Warningf("Could not lookup the current user's information: %s", err)
+	configDir := os.Getenv(configDirEnv)
+	if configDir != "" {
+		return configDir
+	}
 
-		cwd, err := os.Getwd()
-		if err != nil {
-			sylog.Warningf("Could not get current working directory: %s", err)
-			return singularityDir
-		}
+	user, err := user.Current()
+	if err == nil {
+		return filepath.Join(user.HomeDir, singularityDir)
+	}
+	sylog.Warningf("Could not lookup the current user's information: %s", err)
 
+	cwd, err := os.Getwd()
+	if err == nil {
 		return filepath.Join(cwd, singularityDir)
 	}
-
-	return filepath.Join(user.HomeDir, singularityDir)
+	sylog.Warningf("Could not get current working directory: %s", err)
+	
+	return singularityDir
```

Comment:
This is getting a bit complex to review, for what should be an isolated change, because it changes the order in which things are considered / returned. 

Because you are just adding an override with an env var, the safest thing to do here is just insert the block that handles the env var at the top of the function, and leave the rest of the function as-is, with no further changes.

I.E. add...

```
	configDir := os.Getenv(configDirEnv)
	if configDir != "" {
		return configDir
	}
```

... to the top, then leave everything else the same as it was.